src/pybind/mgr/volumes/fs/operations/versions/subvolume_v2.py

```diff
@@ -365,6 +365,7 @@ def safe_to_remove_subvolume_clone(subvol_state):
         return True
 
     def remove(self, retainsnaps=False, internal_cleanup=False):
+        subvol_path = self.path
```

Comment:
```suggestion

```

Reply:
@neesingh-rh This is the culprit for the test failure. During internal cleanup of subvolumes when it's creation is failed, we can't be sure that the metadata is intact. So `self.path` can't be used for internal cleanup.